libecole/include/ecole/dynamics/primalsearch.hpp

```diff
@@ -0,0 +1,40 @@
+#pragma once
+
+#include <map>
+#include <optional>
+
+#include <xtensor/xtensor.hpp>
+
+#include "ecole/dynamics/dynamics.hpp"
+
+namespace ecole::dynamics {
+
+/**
+ * An array of variable identifiers in the transformed problem.
+ */
+using VarIds = xt::xtensor<std::size_t, 1>;
+
+/**
+ * A dictionnary of variable identifiers to variable values.
+ */
+using VarIdVals = std::map<std::size_t, SCIP_Real>;
```

Comment:
So in C++ a `std::map` is (relatively) heavyweight, and we don't use any `map` feature, just convert it to an array of pairs:
```
nonstd::span<std::pair<SCIP_VAR*, SCIP_Real>> const& varvals
```

So why not use a similar thing for the action? For easy Python bindings, we could make these a pair of array, rather than an array of pairs.

Reply:
I agree, I went for an array of pairs later on to have something light. For the action, since Python will give us a map (dictionary) anyway, I though it was just straightforward to use a map in C++ as well, and let pybind do the magic stuff without pain for us. Also the map has a convenient feature: it forbids users to assign the same variable twice to two distinct values. Besides that, I have no strong feelings about keeping a `map` or using something lighter. The lighter the better.